src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -68,7 +68,7 @@ public async Task<UpsertOutcome> UpsertAsync(ResourceWrapper resource, WeakETag
             int etag = 0;
             if (weakETag != null && !int.TryParse(weakETag.VersionId, out etag))
             {
-                throw new ResourceConflictException(weakETag);
+                throw new PreconditionFailedException(string.Format(Core.Resources.ResourceVersionConflict, weakETag?.VersionId));
```

Comment:
I guess one issue is if this is provided but not an integer, we throw, but we don't know what the error should be. We could set the etag to a sentinel value (like -1) and send it though.

Reply:
Oh, good idea. Looking into this now - I think I could potentially get the Cosmos/Sql behavior to match this way.

Reply:
Implemented in 7649951 and updated the [expected outcomes summary sheet](https://microsoft-my.sharepoint.com/:x:/g/personal/rotodd_microsoft_com1/EVWfudFyn4JFnT5o5LJrlNMBig2YlTSSbeRb-0WDmj4DVQ?e=0y4PVh).